src/java.base/share/classes/java/lang/AbstractStringBuilder.java

```diff
@@ -640,14 +640,11 @@ private AbstractStringBuilder appendNull() {
         int count = this.count;
```

Comment:
We should declare `count` before `ensureCapacitiyInternal`. Same for append boolean.

Reply:
> We should declare `count` before `ensureCapacitiyInternal`. Same for append boolean.

Declaring count before ensureCapacityInternal will cause performance regression under x64. It took a lot of time to find this, but the underlying reason is still unclear.